exporters/otlp/otlpmetric/internal/oconf/options.go

```diff
@@ -263,11 +265,32 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
 
 func WithEndpoint(endpoint string) GenericOption {
 	return newGenericOption(func(cfg Config) Config {
-		cfg.Metrics.Endpoint = endpoint
+		// Add scheme if not present
+		if !internal.HasScheme(endpoint) {
+			endpoint = getScheme(cfg) + "://" + endpoint
+		}
+		u, err := url.Parse(endpoint)
+		if err != nil {
+			global.Error(err, "parse url", "input", endpoint)
+			return cfg
+		}
+		cfg.Metrics.Endpoint = u.Host
+		// For OTLP/HTTP endpoint URLs without a per-signal
+		// configuration, the passed endpoint is used as a base URL
+		// and the signals are sent to these paths relative to that.
+		cfg.Metrics.URLPath = path.Join(u.Path, DefaultMetricsPath)
```

Comment:
I don't believe this is correct as gRPC endpoints are not HTTP URLs and should not be treated as such.  This is a general config option that applies to both gRPC and HTTP exporters and cannot include HTTP-specific behavior such as the addition of `DefaultMetricsPath`.

Reply:
Hi @Aneurysm9 , @pellared ,
I am thinking to create two separate `WithEndPoint` functions for HTTP and gRPC and then we can put this code in the HTTP function. I understand about the changes from where this function is being called, will change it accordingly.
Does this seem ok with you?

Reply:
Hi @pellared , 
This is the change for `otlpmetric`. The same changes has been done for `otlptrace` as well. 
But there is no such comment provided in that section of the PR.
So just to confirm, Should I modify the same in `otlptrace` as well?

Reply:
> Should I modify the same in `otlptrace` as well?

Yes 👍